Floofbot/Services/EventHandlerService.cs

```diff
@@ -691,6 +692,34 @@ public Task GuildMemberUpdated(SocketGuildUser before, SocketGuildUser after)
             return Task.CompletedTask;
 
         }
+        public Task HandleWelcomeGate(SocketGuildUser before, SocketGuildUser after) 
+        {
+            var _ = Task.Run(async () =>
+            {
+                if (before.IsPending == after.IsPending) // no welcome gate change
+                    return;
+                FloofDataContext floofDb = new FloofDataContext();
+                var guild = after.Guild;
+                WelcomeGate serverConfig = floofDb.WelcomeGateConfigs.Find(guild.Id);
+
+                if (serverConfig == null || serverConfig.Toggle == false || serverConfig.RoleId == null) // disabled
+                    return;
+
+                try
+                {
+                    var userRole = guild.GetRole((ulong)serverConfig.RoleId);
+                    if (userRole == null)
```

Comment:
May be worth adding an extra debug log here so that it's not a silent fail. Should help with debugging problems with role assignment.